src/entity.unit.ts

```diff
@@ -344,4 +353,70 @@ describe("Entity", () => {
 			expect(movie.toString("[Director]")).toBe("Ridley Scott $100,000.00");
 		});
 	});
+
+	it("setting identifier property of entity marks meta.isNew = false", () => {
```

Comment:
Love the tests! I could read through them and get a sense of what the method is supposed to do and some of the edge cases considered. 👍